code/game/gamemodes/vampire/vampire.dm

```diff
@@ -454,17 +466,31 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 	var/light_available = T.get_lumcount(0.5) * 10
 
 	if(!istype(T))
-		return 0
+		return
 
 	if(!iscloaking)
 		owner.alpha = 255
-		return 0
+		REMOVE_TRAIT(owner, TRAIT_GOTTAGONOTSOFAST, VAMPIRE_TRAIT)
+		return
 
 	if(light_available <= 2)
 		owner.alpha = round((255 * 0.15))
-		return 1
-	else
-		owner.alpha = round((255 * 0.80))
+		ADD_TRAIT(owner, TRAIT_GOTTAGONOTSOFAST, VAMPIRE_TRAIT)
+		return
+
+	REMOVE_TRAIT(owner, TRAIT_GOTTAGONOTSOFAST, VAMPIRE_TRAIT)
+	owner.alpha = round((255 * 0.80))
+
+/datum/vampire/proc/adjust_blood(mob/living/carbon/C, blood_amount = 0)
+	var/unique_suck_id = C.UID()
+	if(!(unique_suck_id in drained_humans))
+		drained_humans[unique_suck_id] = 0
```

Comment:
is there a better way to do this then a list?

Reply:
I'm not sure. Is there?

Reply:
The `if(thing in list)` can be changed to `if(!isnull(list[thing]))` due to it being an assoc list. can't see any other quick improvements though without bloating other objects. The question is if this will be better performant or not though since assoc list checks only pays off after a certain amount of items in the list.
However, after testing it seems that the performance cost is about the same with 10 items in the list. After that the assoc list has a performance advantage. Before that the `t in list` check.

Reply:
The issue is mainly making it hard on a testing basis. On mobile at the moment so can't pull anything to indeprh solutio. Wise out my head

Reply:
I used ancient vampires for testing purposes.